src/main/java/frc/robot/Robot.java

```diff
@@ -440,57 +514,134 @@ private void driveSpeed() {
+    // TODO: Is this ballsInStorage check putting too much trust in the sensor?
+    if (m_controllerDrive.getRawAxis(DriveStation.kIDAxisLT) > 0.50
+        && ballsInStorage < 3)
+      m_motorIntake.set(Constants.kSpeedIntake);
+    else
+      m_motorIntake.set(0);
+
+    // Use this state variable to avoid setting the power of the belt motor more than once.
+    boolean advanceBelt = false;
+    // The digital input returns "true" if the circuit is open. Detecting the
+    // object, the power cell, closes the circuit.
+    boolean ballDetectedEnter = !photoelectricSensorEnter.get();
+    boolean ballDetectedExit = !photoelectricSensorExit.get();
+    // Advance the belt if there's a ball in the enter spot, and more room above.
+    if (ballDetectedEnter) {
+      if (!ballDetectedEnterLastLoop)
+        ++ballsInStorage;
+      if (ballsInStorage < 3)
+        advanceBelt = true;
+      ballDetectedEnterEntry.setBoolean(ballDetectedEnter);
+      ballsInStorageEntry.setDouble(ballsInStorage);
+    } else if (!ballDetectedEnter) {
+      ballDetectedEnterEntry.setBoolean(ballDetectedEnter);
+      advanceBelt = false;
+    }
+    // Keep track of balls exiting.
+    if (!ballDetectedExit) {
+      if (ballDetectedExitLastLoop)
+        --ballsInStorage;
+      // Stop launching the balls if we have finished.
+      if (m_launchBall && ballsInStorage == 0) {
+        m_launchBall = false;
+        m_entryLaunchBall.setBoolean(m_launchBall);
+      }
+      ballDetectedExitEntry.setBoolean(ballDetectedExit);
+      ballsInStorageEntry.setDouble(ballsInStorage);
+    } else if (ballDetectedExit) {
+      ballDetectedExitEntry.setBoolean(ballDetectedExit);
+    }
+
+    // If we are ready to launch the ball, override the false advanceBelt from the storage being full.
+    // TODO: Check to see if the launcher motor has been revved up.
+    if (m_launchBall)
+      advanceBelt = true;
+
+    // If a manipulator bumper is held, disregard all of the previous logic, and force a belt movement.
+    if (m_controllerManip.getRawButton(DriveStation.kIDButtonRB))
+      m_motorBelt.set(Constants.kSpeedBelt);
+    else if (m_controllerManip.getRawButton(DriveStation.kIDButtonLB))
+      m_motorBelt.set(-Constants.kSpeedBelt);
+    // Use the logic based off of the photosensors for belt movement.
+    else
+      m_motorBelt.set(advanceBelt ? Constants.kSpeedBelt : 0);
+
+    // Rev up the launcher motors as soon as we start collecting balls.
+    if (ballsInStorage >= 1)
+      m_motorLauncherLeft.set(Constants.kSpeedLauncher);
+    else
+      m_motorLauncherLeft.set(0);
+
+    ballDetectedEnterLastLoop = ballDetectedEnter;
+    ballDetectedExitLastLoop = ballDetectedExit;
+  }
+
   /**
-   * Determines whether or not the ball can be launched into the power port, and adjusts the robot to
-   * make the shot if it can't.
+   * This function determines whether or not the ball can be launched into the power port, and adjusts
```

Comment:
This is very nit-picky, however in terms of consistency purposes, I believe it would be appropriate to remove "This function."